### ISSUE TITLE
请问能否添加连接被远端强制关闭多次时，将域名加入代理列表的功能？

### ISSUE THREAD

**Issue by showgood163** [cyfdecyf/cow#525]
```
[Error] read tcp 192.168.1.2:57010->125.88.190.22:80: wsarecv: An existing connection was forcibly closed by the remote host.

HTTP Request GET www.52pojie.cn:80/

parse response

Using direct connection.
```

使用cow PAC时的提示如上

```
ERR_CONNECTION_RESET
ERR_CONNECTION_CLOSED
ERR_CONNECTION_TIMED_OUT
```

使用CHROME 直连时的错误码  当然绝不止这些

亲斤 J/ang   伤不起啊

看到作者留了http error code的空位 但是有如下问题

1 错误码必须是整数？
同时填两个（包括以上），不允许；填字符串，也不允许。请问这能修改一下吗？

2 http error code具体是怎么对应的？
能附一个资料来源吗？上面这3个错误提示都是扩展的错误码，找了半天才找到，但好像还是没有什么用处。。

另外再提两个问题：

1 windows 7 ult
COW偶尔会自动崩溃，原因不明，如果需要log请及时联系。

2 ubuntu 16.04 lts
$cow & 启动之后，需要按CTRL+C才能使用同一个终端窗口；但是在CTRL+C之后，$cow & 第一次的进程并没有结束。请问这一块能处理的更好一些吗。。

**Comment by fgggid** [comment]
For the reset problem, I see it is under windows, if you're using the official release 0.9.7 0.9.8 of windows, it is a bug #468 

For the ubuntu 16.04 LTS, I install it as systemd user service;
eg.

```
$ cat /home/ubuntu/cow/cow.service
[Unit]
Description=cow
After=network.target

[Service]
Type=simple
ExecStart=/home/ubuntu/cow/cow
Restart=on-failure
User=ubuntu
Group=ubuntu
UMask=0022

[Install]
WantedBy=multi-user.target

$ sudo systemctl enable /home/ubuntu/cow/cow.service
```

**Comment by showgood163** [comment]
@fgggid 

Thanks for your solution for Ubuntu!! It works like a charm!

But, for windows,
I don't know git or goLang very much and could you write a short tutorial for this?Or you can publish a release in your cow repository.

I think it is needed to be done on github because when I ran /srcipt/build.sh, it told me to get some dependency packs, and after I fetched them and run it again I was told to get more packs on github..


EDIT: I'm now using cow 0.9.6 in both ubuntu and windows.

**Comment by fgggid** [comment]
@showgood163 yes, when building COW, it needs to clone other dependency projects.
And I'm just a C programmer, so I know little about golang build system; I nearly forget how I setup it :P ; If my memory is correct, you can try:
```
$ # install golang
$ sudo apt install golang-go
$ # make working folder - change "~/_prjs/go" to your preference
$ mkdir -p ~/_prjs/go/src
$ cd ~/_prjs/go/src
$ ... clone cow prject ...
$ cd cow
$ # build cow
$ CGO_ENABLED=0 GOPATH=~/_prjs/go go build cow
$ # build windows version
$ CGO_ENABLED=0 GOOS=windows GOARCH=amd64 GOPATH=~/_prjs/go go build cow
```
